### PR TITLE
pd: report write flow through store heartbeat

### PR DESCRIPTION
**Pull request by 5kbpers**
### What is changed and how it works?

Ref #11718 

What's Changed:
* New tick to report write flow to pd worker
* Report write flow through store heartbeat 

kvproto: https://github.com/pingcap/kvproto/pull/845

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
Report write flow through store heartbeat
```

### CODE REVIEW

**Comment by 5kbpers** [comment]
cc @bufferflies

File: components/raftstore/src/store/worker/pd.rs [hunk 1]
```diff
@@ -943,26 +964,36 @@ where
...
             region_peer.last_store_report_read_bytes = region_peer.read_bytes;
             region_peer.last_store_report_read_keys = region_peer.read_keys;
+            region_peer.last_store_report_written_bytes = region_peer.written_bytes;
+            region_peer.last_store_report_written_keys = region_peer.written_keys;
             region_peer
                 .last_store_report_query_stats
                 .fill_query_stats(&region_peer.query_stats);
             if read_bytes < hotspot_byte_report_threshold()
                 && read_keys < hotspot_key_report_threshold()
+                && written_bytes < hotspot_byte_report_threshold()
```
Comment (rleungx) [inline components/raftstore/src/store/worker/pd.rs:980]: `write` should have a different threshold. See https://github.com/tikv/pd/blob/ea57c6d6a0c9ad081aded76545f3d97bc465635f/server/statistics/hot_peer_cache.go#L46-L53

File: components/raftstore/src/store/worker/pd.rs [hunk 2]
```diff
@@ -523,31 +523,49 @@ where
...
 }
 
-const HOTSPOT_KEY_RATE_THRESHOLD: u64 = 128;
-const HOTSPOT_QUERY_RATE_THRESHOLD: u64 = 128;
-const HOTSPOT_BYTE_RATE_THRESHOLD: u64 = 8 * 1024;
+const HOTSPOT_READ_KEY_RATE_THRESHOLD: u64 = 128;
+const HOTSPOT_READ_QUERY_RATE_THRESHOLD: u64 = 128;
+const HOTSPOT_READ_BYTE_RATE_THRESHOLD: u64 = 8 * 1024;
+
+const HOTSPOT_WRITE_KEY_RATE_THRESHOLD: u64 = 32;
```
Comment (lhy1024) [inline components/raftstore/src/store/worker/pd.rs:530]: we also need a threshold for the write query rate, how about 32?

Comment (5kbpers) [reply components/raftstore/src/store/worker/pd.rs:530]: But we don't use it now. Perhaps could add it later

File: components/raftstore/src/store/config.rs [hunk 3]
```diff
@@ -300,7 +301,7 @@ impl Default for Config {
             // Disable consistency check by default as it will hurt performance.
             // We should turn on this only in our tests.
             consistency_check_interval: ReadableDuration::secs(0),
-            report_region_flow_interval: ReadableDuration::minutes(1),
+            report_region_flow_interval: ReadableDuration::secs(1),
```
[the remark this answers is not in the document]
Comment (5kbpers) [reply components/raftstore/src/store/config.rs:304]: This is an interval that raftstore reports write flow to pd worker, change it to 1s for making the flow updated promptly.

File: components/raftstore/src/store/worker/pd.rs (hunk 2 above)
Comment (5kbpers) [reply components/raftstore/src/store/worker/pd.rs:530]: Done

**Pull request closed without merging**